packages/pigeon/lib/generator_tools.dart

```diff
@@ -249,3 +249,42 @@ Map<String, Object> mergeMaps(
   }
   return result;
 }
+
+/// A class name that is enumerated.
+class EnumeratedClass {
+  /// Constructor.
+  EnumeratedClass(this.name, this.enumeration);
+
+  /// The name of the class.
+  final String name;
+
+  /// The enumeration of the class.
+  final int enumeration;
+}
+
+/// Custom codec's custom types are enumerated from 127 down to this number to
+/// avoid collisions with the StandardMessageCodec.
```

Comment:
This isn't avoid collisions, it's violating the documented requirements such that it will *create* collisions. You shouldn't be using anything in this range.

> Type discriminators 0 through 127 inclusive are reserved for use by the base class, as follows:
> - [...]
> - Reserved for future expansion: 14..127
>
> The codec can be extended by overriding this method, calling super for values that the extension does not handle. Type discriminators used by extensions must be greater than or equal to 128 in order to avoid clashes with any later extensions to the base class.

Reply:
Yep, I read this wrong.  I'll change it, good catch.

Reply:
done